C++/C.sublime-syntax

```diff
@@ -13,1427 +14,733 @@ first_line_match: |-
+    - include: enclosure
+  expression.call:
+    - match: \b(?:({{std_funcs}}))\s*(\()
+      captures:
+        1: variable.function.c support.function.std.c
+        2: punctuation.section.group.begin.c
+      push: enclosure.parenthesis
+    - match: \b(?:({{identifier}}))\s*(\()
       captures:
-        1: punctuation.definition.comment.c
+        1: variable.function.c
+        2: punctuation.section.group.begin.c
+      push: enclosure.parenthesis
 
-  line-comments:
-    - match: ^(//) =\s*(.*?)\s*=\s*$\n?
-      scope: comment.line.banner.c
+  function:
+    - include: expression-identifiers
+    - include: operator.pointer
+    - match: ({{identifier}})\s*(\()
       captures:
-        1: punctuation.definition.comment.c
-        2: meta.toc-list.banner.line.c
-    - match: //(?:!|/(?!/))
-      scope: punctuation.definition.comment.c
-      push: line-comment-documentation-body
-    - match: //+
-      scope: punctuation.definition.comment.c
-      push: line-comment-body
+        1: entity.name.function.c
+        2: punctuation.section.group.begin.c
+      push: [function.tail, function.params]
+    - match: \b{{identifier}}\b
+  function.params:
+    - meta_content_scope: meta.function.parameters.c meta.group.c
+    - match: \)
+      scope: punctuation.section.group.end.c
+      pop: 1
+    - match: \}|\] # Fallback
+      scope: invalid.illegal.stray-bracket-end.c
+      pop: 2
+    - match: ; # Fallback
+      scope: invalid.illegal.unexpected-terminator.c
+      pop: 2
+
+    - match: (?={{identifier}}\s*(?![\[,\)]|$))
+      push:
+        - match: \s
+          set: function.params.name
+        - include: expression-identifiers
+    - include: expression
+  function.params.name:
+    - match: (?=[\]\}\);]) # Fallback
+      pop: 1
+    - include: expression-identifiers
+    - match: ({{identifier}})\s*(\[)
+      captures:
+        1: variable.parameter.c
+        2: punctuation.section.brackets.begin
+      set: enclosure.brackets
+    - match: ({{identifier}})\s*(?=[,\)]|$)
+      captures:
+        1: variable.parameter.c
+      pop: 1
+    - include: expression
+  function.tail:
+    - meta_scope: meta.function.c
+    - match: ;
+      pop: 1
+    - include: gnu-attribute
+    - match: \{
+      scope: punctuation.section.block.begin
+      set: function.body
+    - include: expression
+  function.body:
+    - meta_scope: meta.function.c meta.block.c
+    - match: \}
+      scope: punctuation.section.block.end.c
+      pop: 1
+    - include: expression
+    - include: label
 
-  line-comment-body:
-    - meta_include_prototype: false
-    - meta_scope: comment.line.double-slash.c
-    - include: line-comment-end
+  expression-identifiers:
+    - include: operator.word
+    - include: type
+    - include: keyword
+    - include: constant
 
-  line-comment-documentation-body:
-    - meta_include_prototype: false
-    - meta_scope: comment.line.documentation.c
-    - include: line-comment-end
+  ###[SIMPLE STRUCTURES]###
```

Comment:
If you add an extra space on either side of the words, PackageDev will mark as a heading (`###[ SIMPLE STRUCTURES ]###`).

Reply:
Okay, so 79. :sweat_smile: